ktor-network/jvm/src/io/ktor/network/sockets/SocketImpl.kt

```diff
@@ -48,4 +57,13 @@ internal class SocketImpl<out S : SocketChannel>(
     private fun wantConnect(state: Boolean = true) {
         interestOp(SelectInterest.CONNECT, state)
     }
+
+    private fun selfConnect(): Boolean {
+        val localHostAddress = (this.localAddress as? InetSocketAddress)?.address?.hostAddress ?: ""
+        val remoteHostAddress = (this.remoteAddress as? InetSocketAddress)?.address?.hostAddress ?: ""
```

Comment:
Is there a way for `localAddress` to be `null`?

Reply:
Yes, you are right, I will fix this